### ISSUE TITLE
not enough values to unpack (expected 2, got 1)

### ISSUE THREAD

**Issue by andreyto** [vatlab/SoS#910]
Any command in a Bash cell produces the expected standard output, but then prints `not enough values to unpack (expected 2, got 1)`.
Following that, execution of any other cell types (including `sos`) starts generating that same message, until the `sos` kernel is restarted. Variable assignments in `sos` cells no longer work after the Bash error message. Until a Bash cell is executed, Bash magic `!` in `sos` cells works OK, as well as Sos, R, JavaScript and Python3 cell types.
The installation is on CentOS 6.5 with the latest Miniconda3, 64 bit. Tried both installing into a separate Sos environment, as well as into Conda root environment. Tried both the release and Git master head.

**Comment by BoPeng** [comment]
Could you provide more details on how to trigger this bug? I started a sos notebook and entered some simple shell scripts in bash cells and could not reproduce this.

**Comment by andreyto** [comment]
I actually already have a fix. The problem comes from `sos_bash` repo , line 73 of kernel.py. It unconditionally unpacks all elements of a list into two items each, but in my environment, the list ends up containing items like:
```
['BASH_FUNC_module()', '() {  eval `/usr/bin/modulecmd bash $*`\n'], ['}\n'],
```
Apparently, some incorrect splitting of Bash function bodies happens, and some elements have only a single member. I replaced that line of code with:
```
all_vars = {x[0]:(x[1].strip() if len(x)>1 else None) for x in response if x[0].startswith('sos') or x[0] in items}
```
and I am no longer getting an exception.

**Comment by BoPeng** [comment]
Is the following also working?

```diff
diff --git a/src/sos_bash/kernel.py b/src/sos_bash/kernel.py
index 97cdd8b..9e32660 100644
--- a/src/sos_bash/kernel.py
+++ b/src/sos_bash/kernel.py
@@ -68,7 +68,7 @@ class sos_Bash:
     def put_vars(self, items, to_kernel=None):
         # first let us get all variables with names starting with sos
         response = self.sos_kernel.get_response('env', ('stream'))
-        response = [x[1]['text'].split('=', 1) for x in response]
+        response = [x[1]['text'].split('=', 1) for x in response if '=' in x[1]['text']]
         all_vars = {x:y.strip() for x,y in response if x.startswith('sos') or x in items}

         for item in items:
```

**Comment by andreyto** [comment]
Yes. Of course, you are splitting whole function bodies like that, what if they happen to have a `sos` prefix in (some strings)[https://findwords.info/starts/sos] before a `=`?

**Comment by BoPeng** [comment]
Then the function or variable will be automatically passed to other kernels when sos leaves the bash kernel. This kind of automatic data transfer was inspired by beakerX but I personally only use explicit variable exchange.

**Comment by BoPeng** [comment]
Pushed [a patch](https://github.com/vatlab/sos-bash/commit/0802f6191330ed0bf546a20c43411573124a7a6e) and released `sos-bash` 0.9.12.1 for this fix.
